### PR TITLE
Various Example Improvements

### PR DESCRIPTION
**Pull request by ArquintL**
- fixes an issue with ghost erasing interface clauses
- comments out all nearest smaller values as it currently does not terminate on all machines
- changes dense and sparse matrix example to consistently verify on my machine
- adds another test case for buffered channels

Includes changes of PR #210

### CODE REVIEW

File: src/test/resources/regressions/examples/dense_sparse_matrix.gobra [hunk 1]
```diff
@@ -126,17 +126,17 @@ ensures acc(denseMatrix(dm), 1/2) && sparseMatrix(sm, dm.lenX(), dm.lenY())
...
+    fold sparseMatrix(sm, lenX, lenY)
+    invariant acc(denseMatrix(dm), 1/2) && sparseMatrix(sm, lenX, lenY)
+    invariant 0 <= x && x <= lenX && lenX == dm.lenX() && lenY == dm.lenY()
+    invariant forall i, j int :: (0 <= i && i < x && 0 <= j && j < lenY ==> dm.lookup(i, j) == LookupL(sm, lenX, lenY, i, j))
+    for x := 0; x < lenX; x++ {
+        invariant acc(denseMatrix(dm), 1/2) && sparseMatrix(sm, lenX, lenY)
+        invariant 0 <= x && x < lenX && 0 <= y && y <= dm.lenY() && lenX == dm.lenX() && lenY == dm.lenY()
+        invariant forall i, j int :: (0 <= i && i < x && 0 <= j && j < lenY ==> dm.lookup(i, j) == LookupL(sm, lenX, lenY, i, j))
+        invariant forall j int :: (0 <= j && j < y ==> dm.lookup(x, j) == LookupL(sm, lenX, lenY, x, j))
+        for y := 0; y < lenY; y++ {
```
Comment (jcp19) [inline src/test/resources/regressions/examples/dense_sparse_matrix.gobra:139]: Do you have any idea why these changes lead to a more predictable performance/verification on your machine or did you arrive to this version through trial and error?

**Review by jcp19 — APPROVED** [review]
Looks fine to me :)

File: src/test/resources/regressions/examples/dense_sparse_matrix.gobra (hunk 1 above)
Comment (ArquintL) [reply src/test/resources/regressions/examples/dense_sparse_matrix.gobra:139]: My intuition was that it's one indirection less (`lenX` instead of `dm.lenX()`) but I don't know either why this has such an impact on Z3
